### ISSUE TITLE
All active products get fetched from DB

### ISSUE THREAD

**Issue by sauloperez** [openfoodfoundation/openfoodnetwork#4096]
### What we should change and why (this is tech debt)

I haven't found who triggers the query but the Rails log lists

```sql
SELECT "spree_products"."id" FROM "spree_products" WHERE ("spree_products".deleted_at IS NULL)
```

which might be manageable now given our numbers but these could get easily out of hand, specilly when the DB server is busy. This is UK's explain analyze:

```sql
openfoodnetwork=> explain analyze SELECT "spree_products"."id" FROM "spree_products" WHERE ("spree_products".deleted_at IS NULL);
                                                    QUERY PLAN                                                     
-------------------------------------------------------------------------------------------------------------------
 Seq Scan on spree_products  (cost=0.00..1034.07 rows=10596 width=4) (actual time=0.035..6.249 rows=10601 loops=1)
   Filter: (deleted_at IS NULL)
   Rows Removed by Filter: 3513
 Planning time: 0.673 ms
 Execution time: 6.841 ms
```

I don't like that number of rows and the sequential scan :see_no_evil:

### Context

This comes from the investigations done in https://github.com/openfoodfoundation/openfoodnetwork/issues/4077

This query is generated by the `/api/products/bulk_products` endpoint.

It is very similar to https://github.com/openfoodfoundation/openfoodnetwork/issues/4095 but could be considered even worse since it exceeds that one by an order of magnitude.

### Impact and timeline

We better fix it before it causes any major downtime.

**Comment by Matt-Yorkley** [comment]
This might be fixed by pagination in #4081

**Comment by luisramos0** [comment]
adding label performance (performance backlog outside the main 3 epics) and removing from OC page improvement epic as this is a problem in the products  edit page.

**Comment by Matt-Yorkley** [comment]
I bet it's superadmin only, and triggered by `OpenFoodNetwork::Permissions`.

**Comment by Matt-Yorkley** [comment]
Yep, this issue and #4095 both basically require a major refactor of `/lib/open_food_network/permissions.rb`, which could require various corresponding changes throughout the codebase. 

The startling queries are only triggered when logged in with a superuser account though (and have been that way forever), so I'm not sure this is super high priority.

**Comment by sauloperez** [comment]
Understand. We'll eventually get to `permissions.rb` once it becomes the number 1 perf. bottleneck. However, there's no need to do it in one shot. Just splitting that file would be beneficial and allow for other small refactors IMO. The biggest concern I have is that in #4095 we have a query without WHERE...

> We basically need to make sure we paginate everywhere,

totally agree.

**Comment by luisramos0** [comment]
we are closing the performance backlog so I move this one and 4095 to the top of tech backlog backlog.
